pysat/tests/test_instrument_padding.py

```diff
@@ -0,0 +1,641 @@
+    """Unit tests for xarray `pysat.Instrument` with right offset data pad."""
+
+    def setup(self):
+        """Set up the unit test environment for each method."""
+
+        reload(pysat.instruments.pysat_testing_xarray)
+        self.testInst = pysat.Instrument(platform='pysat',
+                                         name='testing_xarray',
+                                         clean_level='clean',
+                                         update_files=True,
+                                         sim_multi_file_right=True,
+                                         pad={'minutes': 5})
+        self.testInst.multi_file_day = True
+        self.ref_time = dt.datetime(2009, 1, 2)
+        self.ref_doy = 2
+        return
+
+    def teardown(self):
+        """Clean up the unit test environment after each method."""
+
+        del self.testInst, self.ref_time, self.ref_doy
+        return
+
+
+class TestMultiFileLeftDataPaddingBasics(TestDataPadding):
+    """Unit tests for pandas `pysat.Instrument` with left offset data pad."""
+
+    def setup(self):
+        """Set up the unit test environment for each method."""
+
+        reload(pysat.instruments.pysat_testing)
+        self.testInst = pysat.Instrument(platform='pysat',
+                                         name='testing',
+                                         clean_level='clean',
+                                         update_files=True,
+                                         sim_multi_file_left=True,
+                                         pad={'minutes': 5})
+        self.testInst.multi_file_day = True
+        self.ref_time = dt.datetime(2009, 1, 2)
+        self.ref_doy = 2
+        return
+
+    def teardown(self):
+        """Clean up the unit test environment after each method."""
+
+        del self.testInst, self.ref_time, self.ref_doy
+        return
+
+
+class TestMultiFileLeftDataPaddingBasicsXarray(TestDataPadding):
+    """Unit tests for xarray `pysat.Instrument` with left offset data pad."""
+
+    def setup(self):
+        """Set up the unit test environment for each method."""
+
+        reload(pysat.instruments.pysat_testing_xarray)
+        self.testInst = pysat.Instrument(platform='pysat',
+                                         name='testing_xarray',
+                                         clean_level='clean',
+                                         update_files=True,
+                                         sim_multi_file_left=True,
+                                         pad={'minutes': 5})
+        self.testInst.multi_file_day = True
+        self.ref_time = dt.datetime(2009, 1, 2)
+        self.ref_doy = 2
+        return
+
+    def teardown(self):
+        """Clean up the unit test environment after each method."""
+
+        del self.testInst, self.ref_time, self.ref_doy
+        return
+
+
+class TestInstListGeneration(object):
+    """Tests to ensure the instrument test class is working as expected."""
```

Comment:
just realized this got moved by accident. This could go with the index tests, or back in the main

Reply:
This should be a part of a different set of instrument tests.

Reply:
moved to a new file